src/main/java/com/walmartlabs/x12/standard/AbstractTransactionSetParserChainable.java

```diff
@@ -0,0 +1,72 @@
+package com.walmartlabs.x12.standard;
+
+import com.walmartlabs.x12.X12Segment;
+import com.walmartlabs.x12.X12TransactionSet;
+
+import java.util.List;
+
+public abstract class AbstractTransactionSetParserChainable implements TransactionSetParser {
+
+    TransactionSetParser nextParser;
+    
+    /**
+     * chainable implementation of the {@link TransactionSetParser} interface
+     * 
+     * if this implementation does not handle the transaction set it will 
+     * pass it on to the next parser in the chain
+     */
+    @Override
+    public X12TransactionSet parseTransactionSet(List<X12Segment> transactionSegments, X12Group x12Group) {
+        if (handlesTransactionSet(transactionSegments, x12Group)) {
+            // we handle this transaction
+            // so parse it
+            return this.doParse(transactionSegments, x12Group);
+        } else if (nextParser != null) {
+            // we don't handle the transaction
+            // try the next link in the chain
+            return nextParser.parseTransactionSet(transactionSegments, x12Group);
+        } else {
+            // if we got here
+            // this transaction type is not handled
+            return null;
```

Comment:
Should this throw an exception or something?

Reply:
Intentionally, this is going to ignore transactions for which a parser is not registered. 
Imagine an X12 document w/ 10 different transaction types, but a user is only interested in one of those. Perhaps some of the transaction types have no implementations. This allows them to be skipped, and the transaction sets that matter to be processed.